conda/resolve.py

```diff
@@ -309,19 +310,19 @@ def _get_strict_channel(self, package_name):
             channel_name = self._strict_channel_cache[package_name] = by_cp[highest_priority]
         return channel_name
 
-    def _broader(self, ms, specs):
+    @memoizemethod
+    def _broader(self, ms, specs_by_name):
         """prevent introduction of matchspecs that broaden our selection of choices"""
-        is_broader = False
-        matching_specs = [s for s in specs if s.name == ms.name]
-
-        if matching_specs and (
-                # is there a version constraint defined for any existing spec, but not MS?
-                (any('version' in _ for _ in matching_specs) and 'version' not in ms)
-                # is there a build constraint on any of the specs, but not on ms?
-                or (any('build' in _ for _ in matching_specs) and 'build' not in ms)
-        ):
-            is_broader = True
-        return is_broader
+        if ms.name not in specs_by_name:
+            return False
```

Comment:
I generally prefer to have a single return from any given function, perhaps aside from an immediate bail  condition just before a long function (sparingly).  It's pretty minor here, but especially when returning multiple values, it can get especially annoying to maintain and debug.

https://stackoverflow.com/a/733858

Fine to keep here - not requesting any changes.